### PR TITLE
add 2.10 release notes

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
adding 2.10 release notes

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by rupal-bq — DISMISSED** [review]

File: release-notes/opensearch-observability.release-notes-2.10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+## Version 2.10.0.0 Release Notes
+
+Compatible with OpenSearch 2.10.0
+
+### Refactoring
+* Fix from upstream core.action changes by @derek-ho in https://github.com/opensearch-project/observability/pull/1590
+* Update backport CI, add PR merged condition by @ps48 in https://github.com/opensearch-project/observability/pull/1587
```
Comment (kavithacm) [inline release-notes/opensearch-observability.release-notes-2.10.0.0.md:7]: This should come under infrastructure, can you change this?

File: release-notes/opensearch-observability.release-notes-2.10.0.0.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## Version 2.10.0.0 Release Notes
+
+Compatible with OpenSearch 2.10.0
+
+### Refactoring
+* Fix from upstream core.action changes by @derek-ho in https://github.com/opensearch-project/observability/pull/1590
```
Comment (kavithacm) [inline release-notes/opensearch-observability.release-notes-2.10.0.0.md:6]: The format for release notes doesn't include PR owner's name in it, so can you remove this as well?

**Comment by kavithacm** [comment]
CI for PR labels is failing, can you fix it? @YANG-DB

File: release-notes/opensearch-observability.release-notes-2.10.0.0.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Version 2.10.0.0 Release Notes
+
+Compatible with OpenSearch 2.10.0
+
+### Infrastructure
+* Update backport CI, add PR merged condition in https://github.com/opensearch-project/observability/pull/1587
```
Comment (kavithacm) [inline release-notes/opensearch-observability.release-notes-2.10.0.0.md:6]: can you pls change it this way? like include PR number in the front of the link
```* Update backport CI, add PR merged condition ([#135](https://github.com/opensearch-project/observability/pull/1587))```

**Review by vamsi-amazon — APPROVED** [review]

File: release-notes/opensearch-observability.release-notes-2.10.0.0.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+### Refactoring
+* Fix from upstream core.action changes in https://github.com/opensearch-project/observability/pull/1590
+* Pull jackson,mockito versions from upstream in https://github.com/opensearch-project/observability/pull/1598
+* Updates demo certs used in integ tests in https://github.com/opensearch-project/observability/pull/1600
```
Comment (derek-ho) [inline release-notes/opensearch-observability.release-notes-2.10.0.0.md:11]: this isn't in 2.10?

**Review by pjfitzgibbons — APPROVED** [review]

**Pull request merged**